src/plugin.js

```diff
@@ -301,6 +302,7 @@ videojs.registerComponent('Overlay', Overlay);
  * @param    {Object} [options={}]
  */
 const plugin = function(options) {
+  const self = this;
```

Comment:
needed to be able to reference `this` in the `add` and `remove` fns, not sure if this is the best way to be able to do that

Reply:
It's a fine practice, though I would suggest calling it something like `player` as that's more meaningful. That way, further down the plugin, future developers don't need to wonder what `self` refers to.